src/job/models/ingestion/seedingJobCreator.ts

```diff
@@ -40,29 +41,31 @@ export class SeedingJobCreator {
           ingestionJobId: ingestionJob.id,
           seedJobType: this.seedJobType,
           seedTaskType,
-          mode,
           layerName,
         });
 
         logger.debug({ msg: 'Getting cache name for layer', layerName });
+
         const cacheName = await this.mapproxyClient.getCacheName({ layerName, cacheType: LayerCacheType.REDIS });
         activeSpan?.addEvent('getCacheName.success', { cacheName });
 
-        const geometry = this.calculateGeometryByMode(mode, ingestionJob);
+        const validCatalogId = internalIdSchema.parse(ingestionJob).internalId;
 
-        if (!geometry) {
-          activeSpan?.addEvent('calculateGeometry.empty');
-          logger.warn({ msg: 'No geometry found, skipping seeding job creation' });
-          return;
-        }
-        activeSpan?.addEvent('calculateGeometry.success', { geometry: JSON.stringify(geometry) });
+        const seedTasks: {
+          type: string;
+          parameters: SeedTaskParams;
+        }[] = [];
 
-        const seedOptions = this.createSeedOptions(mode, geometry, cacheName);
-        activeSpan?.addEvent('createSeedOptions.success', { seedOptions: JSON.stringify(seedOptions) });
+        // Handle different modes
+        const clearTasks = this.handleCleanMode(ingestionJob, cacheName, validCatalogId);
+        const seedModeTasks = this.handleSeedMode(ingestionJob, cacheName, validCatalogId);
```

Comment:
`clearTasks` -> `clearModeTask`
`seedModeTasks` -> `seedModeTask`
singular form naming since as i suggested in another comment it should return a single values